client/credentials.go

```diff
@@ -82,7 +82,11 @@ func (c *Credentials) newAuth(method, url string, h hash.Hash) (*hawk.Auth, erro
 	}
 	if e.Certificate != nil || e.AuthorizedScopes != nil {
 		s, _ := json.Marshal(e)
-		a.Ext = string(s)
+		if string(s) != "{}" {
```

Comment:
`string(s)` can't be `{}` if `e.Certificate != nil || e.AuthorizedScopes != nil`, or am I missing something?